refactor(reports): update balance report

jniles opened the pull request:
This commit updates the balance report with a tree structure, an optional expanded or collapsed presentation and using a fiscal period select component instead of a naked date.

Closes #2617.

This report is now ready for review.

The interface looks like this:

**Configuration Panel**
![balancereportconfiguration](https://user-images.githubusercontent.com/896472/37964686-46dd4ad4-31ba-11e8-8e4b-abd400aaf3a7.PNG)

**View of Combined Debits and Credits**
![combineddebitsandcredits](https://user-images.githubusercontent.com/896472/37964709-503c2320-31ba-11e8-88c6-11066c27d746.PNG)

**View of Separated Debits and Credits**
![separateddebitsandcredits](https://user-images.githubusercontent.com/896472/37964712-50901c1e-31ba-11e8-961a-569a43a786c3.PNG)

jniles committed:
refactor(reports): update balance report

This commit updates the balance report with a tree structure, an
optional expanded or collapsed presentation and using a fiscal period
select component instead of a naked date.

Closes #2617.

jniles committed:
refactor(report): improve balance report

This commit improves the balance report by allowing the user to specify
additional options:
 1. Instead of a date, the user should ask for a period.
 2. Allows the user to filter out zero balances or display them.
 3. Includes a "check row" at the bottom of the balance report to make
 sure that the balance report always balances.
 4. Orders the accounts by account number/hierarchy.
 5. Indents the children under the parents on the balance sheet.

Closes #2617.

mbayopanda commented on client/src/js/components/bhReportPeriodSelect.js:
```diff
@@ -49,7 +49,5 @@ function ReportPeriodSelectController(FiscalPeriod, $filter) {
-  };
-}
\ No newline at end of file
+  $ctrl.onSelect = period => $ctrl.onSelectCallback({ period });
```
👍

mbayopanda commented on server/controllers/finance/reports/balance/index.js:
```diff
@@ -59,164 +56,95 @@ function document(req, res, next) {
+      GROUP BY pt.account_id
+    )s ON a.id = s.account_id
+    ORDER BY a.number;
+  `;
```
💃 good

mbayopanda reviewed (commented):
The code LGTM and the report works well.

I noticed that the totals are not displayed on the bottom of the table.

![balance report](https://user-images.githubusercontent.com/5445251/38189796-814f7072-3659-11e8-9f50-c7dbf86537b9.png)

jniles committed:
fix(balance): fix end to end tests

This commit updates the balance report end to end tests to work
correctly with the new configuration page.
